pandas/core/reshape/pivot.py

```diff
@@ -683,3 +697,35 @@ def _get_names(arrs, names, prefix="row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(names, shared_col_row_names, suffix):
+    keys, counts = _value_counts_arraylike(names, dropna=False)
+    names_count = dict(zip(keys, counts))
+
+    names_mapper = {}
+    unique_names = []
+    # We reverse the names so the numbers are in the order given by the user
+    for name in reversed(names):
```

Comment:
in real-life use cases, is `names` likely to be a large list?

Reply:
What do you consider large? It would be as large as the number of column or rows of the data being crosstabed.

Reply:
What do you consider large? It would be as large as the number of column or rows of the data being crosstabed.

Reply:
It's entirely subjective.  The underlying question is about performance.  The existing factorize functions are cython based and i want to know if this is going to take a significant hit to perf